graphqlapi/src/main/java/io/stargate/graphql/web/GraphqlCache.java

```diff
@@ -85,17 +89,20 @@ public GraphQL getDdl() {
     return ddlGraphql;
   }
 
-  public GraphQL getDml(String keyspace) {
-    DmlGraphqlReference ref = dmlGraphqls.get(keyspace);
+  public GraphQL getDml(String keyspace, Map<String, String> headers) {
+    String decoratedName = persistence.decorateKeyspaceName(keyspace, headers);
```

Comment:
Would it be possible to put the decoration method on `DataStore` to consolidate all user-level access to data in the `DataStore` interface? In the majority of cases rich APIs only need to deal with the `DataStore`. Making `Persistence` a required dependency where keyspaces are manipulated seems a bit inconvenient. Also `DataStore` instances already have relevant request properties. WDYT?

Reply:
If the DataStore-based approach looks reasonable, decorating keyspaces may not be required. We could have something like `DataStore.getNamespace()` and use it here as part of the cache key :thinking:

Reply:
The Graphql cache is global so not sure how to dynamically pass headers many times?

Reply:
Yeah, it might require some refactoring, but I guess we have all the data available at the start of the requests and it is used a bit later to construct the  `DataStore`. @olim7t WDYT?

Reply:
Indeed the cache is global, it serves `GraphQL` instances for all the keyspaces known to this Stargate instance. When an HTTP request comes in, it uses the "external" keyspace such as `ks1`, but the cache knows that keyspace by its "internal" name such as `someNamespace_ks1`.
We need that translation, otherwise we'd either have to: a) have one cache per user, which complicates things unnecessarily, or b) remove the cache and recompute the `GraphQL` for every request, which is impractical.

> use it here as part of the cache key

We already do. The cache key is the decorated keyspace name `someNamespace_ks1`.

> put the decoration method on `DataStore`

It can be done. We'd have to keep a reference to a `DataStore` instance in the cache. For example the "internal" one that we use in the constructor. Line 79:
```java
this.dataStore = dataStoreFactory.createInternal();
```
Line 93:
```java
String decoratedName = this.dataStore.decorateKeyspaceName(keyspace, headers);
```
But it feels wrong to me because it's only a coincidence that we already have that instance available. If we didn't we'd have to create it just for the sake of calling `decorateKeyspace`, which is wasteful. If anything, I think `DataStoreFactory` would be a better place to put it.

Reply:
ok. thanks! let's go with current impl. :+1: